src/shabdawali.js

```diff
@@ -231,20 +231,20 @@ Shabdawali.prototype.typeText = function(cLine){
                 //this.element.textContent  += char;
                 this.element.textContent  = cLine.substr( 0, this.currentLetterIndex  );
                 var that = this;
-                if(char == ' ')
-                {
+                if(char == ' ') {
                     var prevSpaceIndex = cLine.lastIndexOf(' ', this.currentLetterIndex - 2);
                     if(prevSpaceIndex == -1)
                     {
                         prevSpaceIndex = 0;
                     }
                     var prevWord = cLine.substr(prevSpaceIndex, this.currentLetterIndex- prevSpaceIndex);
+                    var wordComplexity = this.wordComplexity(prevWord); 
+                    wordComplexity = wordComplexity == 1? 1 : wordComplexity * (prevWord.length/2);
                     setTimeout(function() {
                         that.typeText(cLine);
-                        }, 50 * prevWord.length );
+                        },this.speed * wordComplexity );
```

Comment:
we'll have to check how long will it pause in case of higher complexity

Reply:
There is a multiplier to the wordComplexity so, higher complexity words will have higher pauses. So, I'm unclear regarding what I have to check?

Reply:
You need not to change anything. Once all the changes are done, I'll check the demo if this behavior looks odd.